### DISCUSSION TITLE
Using ESP32 as graphic card?

### DISCUSSION THREAD

**Discussion by defdefred** [fdivitto/FabGL#149] [Ideas]
Is it stupid to imagine using 1 microcontroller to run a primary application and one ESP32 to run as a dedicated graphic card?

**Comment by OzHawk** [comment]
It is definitely possible.

Check out the link below for an Arduino Nano example of just that.

https://www.instructables.com/Arduino-Basic-PC-With-VGA-Output/

You would be limited by the link speed beteen the primary processing
microcontroller and the VGA microcontroller.

**Comment by defdefred** [comment]
A sort of client-server protocol will be needed... maybe on wifi :-)
I wonder the max resolution/color that could be achieve with a dedicated esp32...

**Reply by niutech** [reply]
You could use VNC protocol, see #97.

**Reply by mtraven** [reply]
I've got an esp32 receiving signals from 2 child board(espNow), making the display & handling user input (keyboard & mouse).  With a 32bit display controller, it fell just short (~30lines) of 640x480.  Bumped it down to 16bit, runs fine.  I should note that my display is all textual, though it contains a LOT of very rapid changes (~100ms).

**Comment by ghost** [comment]
This is exactly how the Agon Light computer - a new eZ80 based machine - works.

It uses an esp32 running FabGL as the "graphics card".  The esp32 and the eZ80 are connected via serial connection.

https://www.thebyteattic.com/p/agon.html
